UnityProject/Assets/Scripts/Tool/AtmosphericAnalyser.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class AtmosphericAnalyser : MonoBehaviour, ICheckedInteractable<HandActivate>
+{
+
+	public bool WillInteract(HandActivate interaction, NetworkSide side)
+	{
+		if (!DefaultWillInteract.Default(interaction, side))
```

Comment:
FYI you don't even need to implement WillInteract if you're just going to call the default validation. If you instead implement IInteractable<HandActivate> you can omit WillInteract and it'll call the default validation automatically.